lib/TextRoles/LinkTextRole.php

```diff
@@ -158,4 +161,67 @@ private function registerLinkFromMatches(array $match, bool $anonymous = false):
 
         return $prev . $id . $next;
     }
+
+    private function replaceStandaloneHyperlinks(string $span): string
+    {
+        // Replace standalone hyperlinks using a modified version of @gruber's
+        // "Liberal Regex Pattern for all URLs", https://gist.github.com/gruber/249502
+        $absoluteUriPattern = '#(?i)\b((?:[a-z][\w\-+.]+:(?:/{1,3}|[a-z0-9%]))('
+            . '?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>'
+            . ']+|(\([^\s()<>]+\)))*\)|[^\s\`!()\[\]{};:\'".,<>?«»“”‘’]))#';
```

Comment:
Just a thought, but it could be cool to use the `x` modifier, that way we can use linebreaks inside the regex instead of doing this ugly concatenation (same below).

Reply:
I did not introduce this regex, just move it. So I would suggest to make it beautiful in a follow up. I am not really experienced with regex so I doubt I can do it...

Reply:
That's fine, I'll do it